core/src/main/java/org/hibernate/ogm/loader/impl/OgmLoader.java

```diff
@@ -488,10 +551,45 @@ private TupleAsMapResultSet getResultSet(Serializable id, OgmLoadingContext ogmL
 		final TupleAsMapResultSet resultset = new TupleAsMapResultSet();
 		if ( getEntityPersisters().length > 0 ) {
 			OgmEntityPersister persister = getEntityPersisters()[0];
-			final EntityKey key = EntityKeyBuilder.fromPersister( persister, id, session );
-			Tuple entry = gridDialect.getTuple( key, persister.getTupleContext() );
-			if ( entry != null ) {
-				resultset.addTuple( entry );
+			if ( useBatching( qp ) ) {
+				// here we expect to receive QueryParameters.positionalParameters full of ids and thus of the same type.
+				// if that's not the case, we are in a bit of a trouble :)
+				// TODO Adding a safeguard but we pay the price for each call, should we remove it?
+				Type idType = persister.getIdentifierType();
+				for (Type wannabeIdType : qp.getPositionalParameterTypes() ) {
+					if ( ! idType.equals( wannabeIdType ) ) {
+						throw new org.hibernate.AssertionFailure( "QueryParameters populated in an unexpected way for Hibernate OGM" );
+					}
+				}
+				int numberOfIds = qp.getPositionalParameterValues().length;
+				// prepare the array of entity keys for each id
+				EntityKey[] keys = new EntityKey[numberOfIds];
+				for ( int index = 0 ; index < numberOfIds ; index++ ) {
+					keys[index] = EntityKeyBuilder.fromPersister( persister, (Serializable) qp.getPositionalParameterValues()[index], session );
+				}
+				if ( multigetGridDialect != null ) {
+					for ( Tuple tuple : multigetGridDialect.getTuples( keys, persister.getTupleContext() ) ) {
+						if ( tuple != null ) {
+							resultset.addTuple( tuple );
+						}
+					}
+				}
+				else {
+					for ( EntityKey entityKey : keys ) {
+						// TODO use the multiload API once introduced
+						Tuple entry = gridDialect.getTuple( entityKey, persister.getTupleContext() );
```

Comment:
Should be doable now, right? I don't see a call to `getTuples()` in the first test method.